python/modules/Stop0l_trigger.py

```diff
@@ -183,108 +183,105 @@ def analyze(self, event):
+        ele_tight = []
+        for ele in electrons:
+                if (ele.pt > 5 and abs(ele.eta) < 2.5 and ele.cutBasedNoIso >= 1 and ele.miniPFRelIso_all < 0.1):
+                        ele_veto.append(ele)
+                        if (ele.cutBasedNoIso >= 3):
+                                ele_mid.append(ele)
+                        if (ele.cutBasedNoIso >= 4):
+                                ele_tight.append(ele)
+        n_ele = len(ele_veto)
+        n_ele_mid = len(ele_mid)
+        n_ele_tight = len(ele_tight)
+
+        zee_mid = []
+        if (n_ele_mid == 2 and ele_mid[0].pt > 40 and ele_mid[1].pt > 20 and (ele_mid[0].charge + ele_mid[1].charge) == 0):
+                zee_cand = ROOT.TLorentzVector()
+                zee_cand = ele_mid[0].p4() + ele_mid[1].p4()
+                #Z mass = 91 GeV
+                #print "zee_cand mass = ", zee_cand.M()
+                if (zee_cand.M() > 81 and zee_cand.M() < 101):
+                        zee_mid.append(zee_cand)
+        n_zee = len(zee_mid)
+                
+        mu_loose = []
+        mu_mid = []
+        for mu in muons:
+                if (mu.pt > 5 and abs(mu.eta) < 2.4 and mu.miniPFRelIso_all < 0.2):
+                        mu_loose.append(mu)
+                        if (mu.mediumId):
+                                mu_mid.append(mu)
+        n_mu = len(mu_loose)
+        n_mu_mid = len(mu_mid)
+
+        zmumu_mid = []
+        if (n_mu_mid == 2 and mu_mid[0].pt > 50 and mu_mid[1].pt > 20 and (mu_mid[0].charge + mu_mid[1].charge) == 0):
+                zmumu_cand = ROOT.TLorentzVector()
+                zmumu_cand = mu_mid[0].p4() + mu_mid[1].p4()
+                #Z mass = 91 GeV
+                #print "zmumu_cand mass = ", zmumu_cand.M()
+                if (zmumu_cand.M() > 81 and zmumu_cand.M() < 101):
+                        zmumu_mid.append(zmumu_cand)
+        n_zmumu = len(zmumu_mid)
 
         photon_loose = []
         photon_mid = []
         for photon in photons:
-                if (abs(photon.eta) < 1.442 or (1.566 < abs(photon.eta) and abs(photon.eta) < 2.5)):
-                        #cutbase =  photon.cutBasedBitmap if self.era != "2016" else photon.cutBased
-                        #if(cutbase >=1):
-                        #        photon_loose.append(photon)
-                        #if(cutbase >=2):
-                        #        photon_mid.append(photon)
-			if self.era == "2016":
-				if photon.cutBased >=1: photon_loose.append(photon)
-				if photon.cutBased >=2: photon_mid.append(photon)
-			else:
-				if bool(photon.cutBasedBitmap & 1): photon_loose.append(photon)
-				if bool(photon.cutBasedBitmap & 2): photon_mid.append(photon)
+                # CMSSW Reference for eta cuts: https://github.com/cms-sw/cmssw/blob/02d4198c0b6615287fd88e9a8ff650aea994412e/DQM/Physics/python/singleTopDQM_miniAOD_cfi.py#L47
+                abeta = math.fabs(photon.eta)
+                if (abeta > 1.4442 and abeta < 1.5660) or (abeta > 2.5):
```

Comment:
Why do you change this? This selects photon in the gap or > 2.5 only!

Reply:
@wang-hui This is the selection you are using?

Reply:
Hi Caleb, I think you are selecting the wrong region.
Could you please change to
if (abs(photon.eta) < 1.4442 or (1.566 < abs(photon.eta) and abs(photon.eta) < 2.5)):

Reply:
Ah sorry... my mistake. I used copy paste from our Photon Stop0l selection which actually vetos photons passing "(abeta > 1.4442 and abeta < 1.5660) or (abeta > 2.5)".

Reply:
It is probably safer to just use the photon Stop0l flag that is already calculated... but I will fix the if statement for now.

Reply:
Here is the Stop0l eta cut I used, but with opposite logic. I will what I put for the trigger selection. 
https://github.com/susy2015/NanoSUSY-tools/blob/dev_v3/python/modules/Stop0lObjectsProducer.py#L150-L151

Reply:
What I wanted to change was 1.442 to 1.4442.

Reply:
Also, I just noticed that this
```(1.566 < abs(photon.eta) and abs(photon.eta) < 2.5)```
should be
```(1.566 > abs(photon.eta) and abs(photon.eta) < 2.5)```

Reply:
Fixed: https://github.com/susy2015/NanoSUSY-tools/commit/824e119a132cfb1530b5772f60cdebc004727df3